docs/design/deployment.adoc

```diff
@@ -5,7 +5,35 @@ v0.1, 2020-05-17
 
 NOTE: This document is under development
 
-== Migration Plane from Openstack Neutron
+== Migration Plan from Openstack Neutron
+
+First, the structures of Neutron models are not consistent with the Alcor's. 
+We need a tool that can transform the existing Neutron network configurations into the Alcor's, so that Alcor can process it.
+These converted network configurations should be handled by Alcor as batch creation operation messages with all parameters specified.
+
+However, we still want users to be able to continue configuring on this openstack environment, 
+as we use the tool to convert the netruon network configurations to Alcors.
+At this time, the virtual switch in compute node side should still be taken over by neutron agents, 
+and an alcor control agent should be in a state that can normally receive configuration, 
+but does not send configuration to the virtual switch.
+The newly generated configuration during this conversion may not be simultaneously converted to Alcor and processed, 
+therefore, we also need to intercept all the operation messages received by the Neutron during the conversion period.
+And in addition to passing them through to the Neutron so that the Neutron can normally process the messages,
+we also need to copy those operation messages and temporarily cache them.
```

Comment:
Let us talk about message forwarding and cache more :-)

In some regions with high QPS, the migration proxy could become the potential bottleneck and we would need carefully plan for that, in terms of latency overhead and size of cached requests. Could you get a rough idea of how much requests could be processed?

Reply:
OK. We could only proxy those write queries. In a region with tens of thousands servers scale, there might be only a few QPS of write queries for network serivces in average, as the QPS of read queries for network serivces could reach to several hundreds.

Reply:
> OK. We could only proxy those write queries. 

Nice. If the "upper" API Gateway could selectively sends the write quests to the new "Neutron migration proxy", then problem solved :-)